BrainPortal/app/models/boutiques_cluster_task.rb

```diff
@@ -98,6 +98,19 @@ def setup #:nodoc:
     true
   end
 
+  # narrows down local dp paths only to the most relevant, to be used at setup
+  def local_dp_storage_paths
+    return super & input_file_dps.pluck(:remote_dir)
+  end
+
+  # lists the dp involved into input files
+  def input_file_dps
+    file_ids = descriptor_for_setup.file_inputs.map do |input|
+      invoke_params[input.id]
+    end
+    return Userfile.where(id: file_ids).map &:data_provider
```

Comment:
You are returning a list with duplicate data providers in it (e.g. when two input files are on the same DP)

Reply:
I never claimed it returns unique dp(s).

Reply:
But now it returns unique dp